graphql/middleware.go

```diff
@@ -0,0 +1,36 @@
+package graphql
+
+import "context"
+
+type ComputationInput struct {
+	Id          string
+	Query       string
+	ParsedQuery *Query
+	Variables   map[string]interface{}
+	Ctx         context.Context
+	Previous    interface{}
+}
+
+type ComputationOutput struct {
+	Metadata map[string]interface{}
+	Current  interface{}
+	Error    error
+}
+
+type MiddlewareFunc func(input *ComputationInput, output *ComputationOutput, next NextFunc)
```

Comment:
What do you think of making MiddlewareFunc and NextFunc return *ComputationOutput instead?